### DISCUSSION TITLE
pyvista cannot be installed properly (multiple approaches tried)

### DISCUSSION THREAD

**Discussion by songhanzhang** [pyvista/pyvista#6566] [Q&A]
### Describe the bug, what's wrong, and what you expected.

I'm using Python 3.12.1 on Mac OS. I tried to install pyvista from three approaches:

1. using pip

pip install pyvista
pip3 install pyvista

and it returns:

Requirement already satisfied: pyvista in /opt/miniconda3/lib/python3.12/site-packages (0.44.1)
Requirement already satisfied: matplotlib>=3.0.1 in /opt/miniconda3/lib/python3.12/site-packages (from pyvista) (3.9.2)
...
Requirement already satisfied: certifi>=2017.4.17 in /opt/miniconda3/lib/python3.12/site-packages (from requests>=2.19.0->pooch->pyvista) (2024.6.2)

But when using in python: import pyvista as pv, then it says:

ModuleNotFoundError: No module named 'pyvista'

Fail ...

2. using Miniconda

I tried to install again using Miniconda (in a created environment):

conda install pyvista

Channels:
 - defaults
Platform: osx-arm64
Collecting package metadata (repodata.json): done
Solving environment: failed

PackagesNotFoundError: The following packages are not available from current channels:

  - pyvista

Current channels:

  - defaults

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.

Again, fail ...

3. using Miniconda 

conda install -c conda-forge pyvista

as show on the pyvista website, and it can be installed successfully. But when running the line:

>> grid = pv.UnstructuredGrid(cells, np.array([12]), ele_vertices)

it gives error message:

Traceback (most recent call last):
  File "/Users/Documents/Python/TFEA/ex_pipe.py", line 147, in <module>
    grid = pv.UnstructuredGrid(cells, np.array([12]), ele_vertices)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/miniconda3/envs/floatfarm/lib/python3.12/site-packages/pyvista/core/pointset.py", line 1834, in __init__
    self._from_arrays(args[0], args[1], args[2], deep, **kwargs)
  File "/opt/miniconda3/envs/floatfarm/lib/python3.12/site-packages/pyvista/core/pointset.py", line 1945, in _from_arrays
    vtkcells = CellArray(cells)
               ^^^^^^^^^^^^^^^^
...

Again, fail ...

I have being strictly following all the steps, while I cannot exactly find out the reason why none of the approaches works. The remaining packages e.g., numpy, scipy ... can all be installed and used successfully from Miniconda. Is there anything else specially required for pyvista?

### Steps to reproduce the bug.

```python
Insert code here
```

### System Information

```shell
ModuleNotFoundError: No module named 'pyvista'
```

**Comment by tkoyama010** [comment]
We recommend using venv to install libraries in Python. Below is an example of installation.
```bash
$ python -m venv
$ . venv/bin/activate
(venv) pip install pyvista
(venv) python
>>> import pyvista as pv
```

**Reply by banesullivan** [reply]
+1, this log looks like the you are using the base conda environment, mixing in pip, and probably accidentally mixing in a system `pip3` that came from somewhere else. I recommend you read up on the best practices/tutorials for managing conda environments prior to debugging a specific package installation: https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html

Further, if using conda to install pyvista and VTK you *must* use the `conda-forge` channel, not the `defaults` channel

Hope this helps
